src/Components/WebAssembly/WebAssembly/src/Services/LazyAssemblyLoader.cs

```diff
@@ -99,7 +105,7 @@ private async Task<IEnumerable<Assembly>> LoadAssembliesInClientAsync(IEnumerabl
                 // into the default app context.
                 var loadedAssembly = AssemblyLoadContext.Default.LoadFromStream(new MemoryStream(assembly));
                 loadedAssemblies.Add(loadedAssembly);
-                _loadedAssemblyCache.Add(loadedAssembly);
+                _loadedAssemblyCache.Add(loadedAssembly.GetName().Name + ".dll", loadedAssembly);
```

Comment:
Is there any chance of errors with clashing dictionary keys here? Or in other words, is there any possibility of a race condition where the same assembly gets loaded twice in parallel (e.g., due to edge cases like the developer having two separate `<Router>` components), which would normally work out fine except because of this line?

Might be worth changing to:

```suggestion
                _loadedAssemblyCache[loadedAssembly.GetName().Name + ".dll"] = loadedAssembly;
```

Reply:
There is a chance. This ends up happening in the `ToDictionary` call in the constructor but it's also likely to happen here if the user attempts to lazy-load something twice. Our other checks should safe-guard against this but better safe than sorry.